snippets/unflattenObject.md

```diff
@@ -5,29 +5,27 @@ tags: object,advanced
 
 Unflatten an object with the paths for keys.
 
-- Use `Object.keys(obj)` combined with `Array.prototype.reduce()` to convert flattened path node to a leaf node.
-- If the value of a key contains a dot delimiter (`.`), use `Array.prototype.split('.')`, string transformations and `JSON.parse()` to create an object, then `Object.assign()` to create the leaf node.
-- Otherwise, add the appropriate key-value pair to the accumulator object.
+- Iterate over the object keys using `Object.keys(obj)` and `Array.prototype.forEach()`.
+- Split each key with a dot delimiter (`.`) using `Array.prototype.split('.')`and use `Array.prototype.reduce()` to convert flattened paths to leaf node.
+- If the current accumulator already contains value against a particular key, return its value as the next accumulator.
+- Otherwise, add the appropriate key-value pair to the accumulator object and return value as the accumulator.
+
 
 ```js
-const unflattenObject = obj =>
-  Object.keys(obj).reduce((acc, k) => {
-    if (k.indexOf('.') !== -1) {
-      const keys = k.split('.');
-      Object.assign(
-        acc,
-        JSON.parse(
-          '{' +
-            keys.map((v, i) => (i !== keys.length - 1 ? `"${v}":{` : `"${v}":`)).join('') +
-            obj[k] +
-            '}'.repeat(keys.length)
-        )
-      );
-    } else acc[k] = obj[k];
-    return acc;
-  }, {});
+const unflattenObject = obj =>{
+    var result = {}
+    Object.keys(obj).forEach((k)=>{
+      var keys = k.split('.')
+      keys.reduce(function(acc, e, i) {
```

Comment:
I think you can move `k.split(.)` in this line as the caller and use the fourth argument in the callback to access it like so and remove the previous line:
```suggestion
      k.split('.').reduce(function(acc, e, i, keys) {
```